get-live-images.sh

```diff
@@ -0,0 +1,249 @@
+#!/bin/bash -xe
+
+# Check and set http(s)_proxy. Required for cURL to use a proxy
+export http_proxy=${http_proxy:-$HTTP_PROXY}
+export https_proxy=${https_proxy:-$HTTPS_PROXY}
+export no_proxy=${no_proxy:-$NO_PROXY}
+export CURL_CA_BUNDLE=${CURL_CA_BUNDLE:-/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem}
+
+# Which image should we use
+export RHCOS_IMAGE_URL=${1:-$RHCOS_IMAGE_URL}
+if [ -z "$RHCOS_IMAGE_URL" ] ; then
+    echo "No image URL provided"
+    exit 1
+fi
+
+function download_image() {
+	CONNECT_TIMEOUT=120
+	MAX_ATTEMPTS=5
+
+	for i in $(seq ${MAX_ATTEMPTS}); do
+		if ! curl -g --compressed -L --connect-timeout ${CONNECT_TIMEOUT} -o "$1" "$2"; then
+			if (( ${i} == ${MAX_ATTEMPTS} )); then
+				echo "Download failed."
+				exit 1
+			else
+				SLEEP_TIME=$((i*i))
+				echo "Download failed, retrying after ${SLEEP_TIME} seconds..."
+				sleep ${SLEEP_TIME}
+			fi
+		else
+			break
+		fi
+	done
+}
+
+function get_provisioning_interface() {
+	if [ -n "${PROVISIONING_INTERFACE}" ]; then
+		# don't override the PROVISIONING_INTERFACE if one is provided
+		echo ${PROVISIONING_INTERFACE}
+		return
+	fi
+
+	local interface="provisioning"
+	for mac in ${PROVISIONING_MACS//,/ } ; do
+		if ip -br link show up | grep -q "$mac"; then
+			interface=$(ip -br link show up | grep "$mac" | cut -f 1 -d ' ')
+			break
+		fi
+	done
+	echo $interface
+}
+
+# Wait for the interface or IP to be up, sets $IRONIC_IP
+function wait_for_interface_or_ip() {
+	# If $PROVISIONING_IP is specified, then we wait for that to become available on an interface, otherwise we look at $PROVISIONING_INTERFACE for an IP
+	if [ ! -z "${PROVISIONING_IP}" ]; then
+		export IRONIC_IP=$(printf %s "${PROVISIONING_IP}" | sed -e 's%/.*%%')
+		until ip -br addr show | grep -q -F " ${IRONIC_IP}/"; do
+			echo "Waiting for ${IRONIC_IP} to be configured on an interface"
+			sleep 1
+		done
+	else
+		until [ ! -z "${IRONIC_IP}" ]; do
+			echo "Waiting for ${PROVISIONING_INTERFACE} interface to be configured"
+			export IRONIC_IP=$(ip -br add show scope global up dev "${PROVISIONING_INTERFACE}" | awk '{print $3}' | sed -e 's%/.*%%' | head -n 1)
+			sleep 1
+		done
+	fi
+
+	# If the IP contains a colon, then it's an IPv6 address, and the HTTP
+	# host needs surrounding with brackets
+	if [[ "$IRONIC_IP" =~ .*:.* ]]; then
+		export IPV=6
+		export IRONIC_URL_HOST="[$IRONIC_IP]"
+	else
+		export IPV=4
+		export IRONIC_URL_HOST=$IRONIC_IP
+	fi
+}
+
+function iso_embed() {
```

Comment:
I'd prefer we do this in ironic-image to avoid duplicating all the parameters and the IP address logic.

Reply:
> I'd prefer we do this in ironic-image to avoid duplicating all the parameters and the IP address logic.

TBH, I only added this in the last minute and wanted to discuss this with you and @hardys as an alternative to the ironic-image logic. This is actually a minimal change to the rhcos-downloader container as we have to do was add 2-3 extra env variables. I also feel its a more apt place because any rhcos-based logic can be separated from ironic.